spring-session/src/main/java/org/springframework/session/hazelcast/PrincipalNameExtractor.java

```diff
@@ -34,42 +32,16 @@
  */
 public class PrincipalNameExtractor extends ValueExtractor<MapSession, String> {
 
-	private static final PrincipalNameResolver PRINCIPAL_NAME_RESOLVER =
-			new PrincipalNameResolver();
+	private final SingleIndexResolver<MapSession> indexResolver =
+			new PrincipalNameIndexResolver<MapSession>();
 
 	@SuppressWarnings("unchecked")
 	public void extract(MapSession target, String argument,
 			ValueCollector collector) {
-		String principalName = PRINCIPAL_NAME_RESOLVER.resolvePrincipal(target);
+		String principalName = this.indexResolver.resolveIndexValueFor(target);
```

Comment:
This would benefit from `IndexResolver` having a method to resolve single index. ATM I'm using `SingleIndexResolver` directly here, which I had to change a bit (make `SingleIndexResolver#resolveIndexValueFor` `public` instead of `protected` as originally outlined). `ValueExtractor`s are instantiated by Hazelcast, so perhaps the best would be to have Hazelcast specific `IndexResolver` implementation.